app/browser/api/ledger.js

```diff
@@ -0,0 +1,2187 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const acorn = require('acorn')
+const moment = require('moment')
+const Immutable = require('immutable')
+const electron = require('electron')
+const ipc = electron.ipcMain
+const path = require('path')
+const os = require('os')
+const qr = require('qr-image')
+const underscore = require('underscore')
+const tldjs = require('tldjs')
+const urlFormat = require('url').format
+const queryString = require('queryString')
+const levelUp = require('level')
+const random = require('random-lib')
+const uuid = require('uuid')
+
+// Actions
+const appActions = require('../../../js/actions/appActions')
+
+// State
+const ledgerState = require('../../common/state/ledgerState')
+const pageDataState = require('../../common/state/pageDataState')
+
+// Constants
+const settings = require('../../../js/constants/settings')
+const messages = require('../../../js/constants/messages')
+
+// Utils
+const tabs = require('../../browser/tabs')
+const locale = require('../../locale')
+const appConfig = require('../../../js/constants/appConfig')
+const getSetting = require('../../../js/settings').getSetting
+const {fileUrl, getSourceAboutUrl} = require('../../../js/lib/appUrlUtil')
+const urlParse = require('../../common/urlParse')
+const ruleSolver = require('../../extensions/brave/content/scripts/pageInformation')
+const request = require('../../../js/lib/request')
+const ledgerUtil = require('../../common/lib/ledgerUtil')
+
+// Caching
+let locationDefault = 'NOOP'
+let currentUrl = locationDefault
+let currentTimestamp = new Date().getTime()
+let visitsByPublisher = {}
+let bootP
+let quitP
+let notificationPaymentDoneMessage
+const _internal = {
+  verboseP: true,
```

Comment:
These items (verboseP, etc) should be accessible outside the file (or should have another way to override them; ex: read a preference). When running unit tests against these, there's a lot of output

Reply:
done